docassemble/assemblylinewizard/test_map_names.py

```diff
@@ -85,52 +49,151 @@ def test_mapped_scenarios(self, run_from_yaml=False):
-  "court": "str(courts[0])",
-  "other_party": "str(other_parties[0])",
-  "child": "str(children[0])",
-  "guardian_ad_litem": "str(guardians_ad_litem[0])",
-  "witness": "str(witnesses[0])",
-  "users": "str(users)",
-  "plaintiffs": "str(plaintiffs)",
-  "defendants": "str(defendants)",
-  "petitioners": "str(petitioners)",
-  "respondents": "str(respondents)",
-  "spouses": "str(spouses)",
-  "parents": "str(parents)",
-  "guardians": "str(guardians)",
-  "caregivers": "str(caregivers)",
-  "attorneys": "str(attorneys)",
-  "translators": "str(translators)",
-  "debt_collectors": "str(debt_collectors)",
-  "creditors": "str(creditors)",
-  "courts": "str(courts)",
-  "other_parties": "str(other_parties)",
-  "children": "str(children)",
-  "guardians_ad_litem": "str(guardians_ad_litem)",
-  "witnesses": "str(witnesses)",
+  "user": "users[0]",
+  "plaintiff": "plaintiffs[0]",
+  "defendant": "defendants[0]",
+  "petitioner": "petitioners[0]",
+  "respondent": "respondents[0]",
+  "spouse": "spouses[0]",
+  "parent": "parents[0]",
+  "guardian": "guardians[0]",
+  "caregiver": "caregivers[0]",
+  "attorney": "attorneys[0]",
+  "translator": "translators[0]",
+  "debt_collector": "debt_collectors[0]",
+  "creditor": "creditors[0]",
+  "court": "courts[0]",
+  "other_party": "other_parties[0]",
+  "child": "children[0]",
+  "guardian_ad_litem": "guardians_ad_litem[0]",
+  "witness": "witnesses[0]",
+  "users": "users",
+  "plaintiffs": "plaintiffs",
+  "defendants": "defendants",
+  "petitioners": "petitioners",
+  "respondents": "respondents",
+  "spouses": "spouses",
+  "parents": "parents",
+  "guardians": "guardians",
+  "caregivers": "caregivers",
+  "attorneys": "attorneys",
+  "translators": "translators",
+  "debt_collectors": "debt_collectors",
+  "creditors": "creditors",
+  "courts": "courts",
+  "other_parties": "other_parties",
+  "children": "children",
+  "guardians_ad_litem": "guardians_ad_litem",
+  "witnesses": "witnesses",
 
   # Starts with no names
   "docket_number": "docket_numbers[0]",
-  "docket_numbers": "str(docket_numbers)",
+  "docket_numbers": "docket_numbers",
+  "signature_date": "signature_date",
+
+  # Reserved start with unreserved end
+  #"user_address_street2_zip": "users[0].address_street2_zip",
+  "user_address_street2_zip": "user_address_street2_zip",  
+
+  # Not reserved
+  "my_user_name_last": "my_user_name_last",
+  "foo": "foo",
+}
+
+interview_order_scenarios = {
+  # Reserved whole words
+  "signature_date": "signature_date",
+  "attorney_of_record_address_on_one_line": "attorney_of_record_address_on_one_line",
+
+  # Reserved endings
+  "user1": "users.gather()",
+  "user2": "users.gather()",
+  "user__2": "users.gather()",
+  "user_name": "users.gather()",
+  "user_name_full": "users.gather()",
+  "user_name_first": "users.gather()",
+  "user_name_middle": "users.gather()",
+  "user_name_last": "users.gather()",
+  "user_name_suffix": "users.gather()",
+  "user_gender": "users.gather()",
```

Comment:
So sorry that I totally missed one of the requested review items. I believe this should turn into `users[0].gender`. We're not currently building a block that makes sure to gather all of an objects attributes in the way you showed from the docs. We have to rely on doing them one by one.

Not completely sure, so feel free to weigh in, @nonprofittechy

Reply:
What @plocket says sounds correct to me